app/Config.hs

```diff
@@ -97,8 +97,11 @@ instance FromJSON Config where
     let first Target{..} = host
         configHosts = nubBy ((==) `on` first)
           (maybeToList (Target <$> host <*> pure port <*> pure shell <*> pure sshArgs) ++ hs)
-    configRepo       <- o .: "repo"
-    configRevision   <- o .: "revision"
+        getRepository m =
+              LocalDirectory <$> m .: "local_directory"
+          <|> (m .: "external-repository" >>= \m' -> ExternalRepository <$> m' .: "url" <*> m' .: "revision")
```

Comment:
I don't think this should be necessary for now and should be safe to remove it. At least not until adding support for more sources like mercury, etc.

Reply:
Which one you mean it's not necessary? The `external-repository` option?

Reply:
I think we can safely remove this whole line until we add support for more sources